### PR TITLE
virtio_fs_hotplug: 'session' needs to be re-obtained after vm reboot

### PR DESCRIPTION
**Pull request by nickzhq**
ID: 1990367

Signed-off-by: Houqi (Nick) Zuo hzuo@redhat.com

### CODE REVIEW

**Comment by nickzhq** [comment]
Command: `python3 ConfigTest.py --testcase=virtio_fs_hotplug.plug.one_fs_source --guestname=Win2019 --machines=q35 --customsparams="cdrom_virtio = isos/windows/virtio-win-prewhql-0.1-215.iso" --firmware=default_bios --clone=yes`

(1/3) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.q35: STARTED
/ (1/3) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.q35: PASS (1388.40 s)
(2/3) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_fs_hotplug.plug.one_fs_source.q35: STARTED
\ (2/3) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_fs_hotplug.plug.one_fs_source.q35: ERROR: argument must be an int, or have a fileno() method. (258.71 s)
(3/3) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_fs_hotplug.plug.one_fs_source.with_repetition.q35: STARTED
- (3/3) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_fs_hotplug.plug.one_fs_source.with_repetition.q35: PASS (1871.30 s)
RESULTS    : PASS 2 | ERROR 1 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0

The error at here is related to another bug, which is without any connections in this pr.
This pr is related to the third case whose status is PASS.

**Comment by nickzhq** [comment]
@xiagao @YongxueHong  Please check this pr. Thanks a lot.

File: qemu/tests/virtio_fs_hotplug.py [hunk 1]
```diff
@@ -141,11 +141,15 @@ def start_vfs_service(session):
                     if s:
                         test.fail("Fail command: %s. Output: %s" % (reg_cmd, o))
                 error_context.context("Reboot guest.", test.log.info)
-                session = vm.reboot()
+                vm.reboot()
+                # After reboot, 'session' is changed.
+                # The 'session' needs to be obtained from vm
+                session = vm.wait_for_login()
```
Comment (zhencliu) [inline qemu/tests/virtio_fs_hotplug.py:147]: we don't need to log into vm again for vm.reboot will return a new session object, we can use that.

**Comment by xiagao** [comment]
wget https://github.com/autotest/tp-qemu/pull/3332.patch
git am 3332.patch
python3 ConfigTest.py --testcase=virtio_fs_hotplug.plug.one_fs_source --guestname=Win2019 --machines=q35 --customsparams="cdrom_virtio = isos/windows/virtio-win-prewhql-0.1-221.iso" --firmware=default_bios
 (1/3) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.q35: PASS (1759.27 s)
 (2/3) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_fs_hotplug.plug.one_fs_source.q35: ERROR: argument must be an int, or have a fileno() method. (317.40 s)
 (3/3) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_fs_hotplug.plug.one_fs_source.with_repetition.q35: ERROR: Shell process terminated while waiting for command to complete: "wmic logicaldisk where (VolumeName='myfs') get DeviceID"    (status: 1,    output: '') (213.61 s)
----->Still hit this issue.

**Comment by nickzhq** [comment]
Here is Linux version
(1/1) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.virtio_fs_hotplug.plug.one_fs_source.with_repetition.q35: STARTED
/ (1/1) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.virtio_fs_hotplug.plug.one_fs_source.with_repetition.q35: PASS (1662.51 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0

**Comment by xiagao** [comment]
(1/3) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.q35: PASS (1149.41 s)
 (2/3) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.virtio_fs_hotplug.plug.one_fs_source.q35: ERROR: argument must be an int, or have a fileno() method. (281.58 s)
 (3/3) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.virtio_fs_hotplug.plug.one_fs_source.with_repetition.q35: PASS (1613.97 s)

LGTM
ack.

**Comment by zhencliu** [comment]
> (1/3) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.q35: STARTED / (1/3) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.q35: PASS (1388.40 s) (2/3) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_fs_hotplug.plug.one_fs_source.q35: STARTED \ (2/3) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_fs_hotplug.plug.one_fs_source.q35: ERROR: argument must be an int, or have a fileno() method. (258.71 s) (3/3) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_fs_hotplug.plug.one_fs_source.with_repetition.q35: STARTED
> 
> * (3/3) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_fs_hotplug.plug.one_fs_source.with_repetition.q35: PASS (1871.30 s)
>   RESULTS    : PASS 2 | ERROR 1 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
> 
> The error at here is related to another bug, which is without any connections in this pr. This pr is related to the third case whose status is PASS.

Agreed. For the new issue, please file a new bz to track it, and fix it there.

**Comment by zhencliu** [comment]
LGTM

**Comment by nickzhq** [comment]
Here are the whole cases of virtio_fs_hotplug with guest Win2019
 
(1/4) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.q35: STARTED
/ (1/4) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.q35: PASS (1042.47 s)
 (2/4) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_fs_hotplug.plug.one_fs_source.q35: STARTED
\ (2/4) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_fs_hotplug.plug.one_fs_source.q35: ERROR: argument must be an int, or have a fileno() method. (258.89 s)
 (3/4) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_fs_hotplug.plug.one_fs_source.with_repetition.q35: STARTED
- (3/4) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_fs_hotplug.plug.one_fs_source.with_repetition.q35: PASS (1863.51 s)
 (4/4) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_fs_hotplug.unplug.one_fs_source.q35: STARTED
| (4/4) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_fs_hotplug.unplug.one_fs_source.q35: PASS (137.82 s)
RESULTS    : PASS 3 | ERROR 1 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0

**Pull request merged**